### PR TITLE
Searchable keycodes

### PR DESCRIPTION
**Pull request by yanfali**
Proof of Concept for searchable keycodes. @noroadsleft could use some styling input

### CODE REVIEW

**Comment by yanfali** [comment]
> I'm actually 99% okay with how you've styled this. It would be nice if the search field looked more like the other UI elements (border style, padding, etc.). I had another thought though.
> 
> Could we use key descriptions/titles here and also make them searchable, so if a user searches for `Fn`, it could highlight (well, not dim) the `MO()` key?

I think it already uses titles, would be pretty easy to add one more search field. Try it

**Comment by noroadsleft** [comment]
It's a small change:

```diff
diff --git a/src/components/Keycodes.vue b/src/components/Keycodes.vue
index b8384f3f6..fd627b39f 100644
--- a/src/components/Keycodes.vue
+++ b/src/components/Keycodes.vue
@@ -170,6 +170,11 @@ export default {
 .end-tab {
   grid-column: -1;
 }
+.end-tab input {
+  padding: 3px 7px;
+  border: 1px solid #cdcdcd;
+  border-radius: 4px;
+}
 .tab-area {
   height: 350px;
   padding: 10px 5px;
```

File: src/components/Keycodes.vue [hunk 1]
```diff
@@ -116,6 +125,19 @@ export default {
       this.clearTimeout = window.setTimeout(() => {
         store.commit('app/setMessage', '');
       }, 3000);
+    },
+    filterClass(key) {
+      if (this.searchFilter === '' || isUndefined(key.code)) {
```
Comment (zekth) [inline src/components/Keycodes.vue]: Why not `if (!this.searchFilter` ?

Comment (yanfali) [reply src/components/Keycodes.vue]:
```
s=''
""
!s
true
```

Comment (yanfali) [reply src/components/Keycodes.vue]: I think this is counter intuitive and harder to read. It would only be slightly better if it were !!s

**Comment by yanfali** [comment]
@noroadsleft that styling work for you?

Screen Shot 2019-11-23 at 21 59 31

**Comment by noroadsleft** [comment]
I like the styling but I had another thought:

Is there any way the search could modify the tab labels with the number of keycode results under that section?

Mock-up screenshot because the App tab is too narrow:

image

**Comment by yanfali** [comment]
> I like the styling but I had another thought:
> 
> Is there any way the search could modify the tab labels with the number of keycode results under that section?
> 
> Mock-up screenshot because the App tab is too narrow:
> 
> image

Yeah I thought about it. I just kept it simple for now.

**Comment by noroadsleft** [comment]
We can do it later, that's fine.

**Comment by yanfali** [comment]
Screen Shot 2019-11-25 at 12 32 09

@noroadsleft

File: src/components/Keycodes.vue [hunk 2]
```diff
@@ -145,6 +185,17 @@ export default {
   cursor: pointer;
   margin-bottom: -1px;
 }
+.end-tab {
```
Comment (noroadsleft) [inline src/components/Keycodes.vue]: Makes it look like there's a space between the tab name and the result count.

```suggestion
.tab span {
  margin-left: 4px;
}
.end-tab {
```

File: src/components/Keycodes.vue [hunk 3]
```diff
@@ -25,7 +25,7 @@
           v-for="(key, index) in keycodesByGroup"
           :key="index"
           @click="changeActive(index)"
-          :title="index"
+          :title="$t('message.keycodesTab.' + index + '.label')"
```
Comment (noroadsleft) [inline src/components/Keycodes.vue]: This has driven me nuts for weeks. :heart:

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**